File: server/kotlin/src/main/kotlin/com/lebonpoint/application/usecases/CreateItemUseCase.kt
package com.lebonpoint.application.usecases

import com.lebonpoint.application.services.ValidationService
import com.lebonpoint.domain.entities.CreateItemData
import com.lebonpoint.domain.entities.Item
import com.lebonpoint.domain.repositories.ItemRepository
import com.lebonpoint.shared.ValidationException

/**
 * Use case for creating a new item
 */
class CreateItemUseCase(
    private val repository: ItemRepository
) {
    /**
     * Execute the use case
     * @param data Item creation data
     * @return Created item
     * @throws ValidationException if validation fails
     */
    suspend fun execute(data: CreateItemData): Item {
        // Validate input
        ValidationService.validateTitle(data.title)
        ValidationService.validatePriceCents(data.priceCents)
        ValidationService.validateCondition(data.condition)
        ValidationService.validateStatus(data.status)

        // Create item
        return repository.create(data)
    }
}

File: server/typescript/src/application/use-cases/create-item.use-case.ts
import type { IItemRepository } from '../../domain/repositories/item.repository.port.js';
import type { CreateItemData } from '../../domain/entities/item.entity.js';
import { createItemWithDefaults } from '../../domain/entities/item.entity.js';
import { ValidationService } from '../services/validation.service.js';
import type { Item } from '../../domain/entities/index.js';
import { Condition, Status } from '../../domain/entities/index.js';

/**
 * Create item use case
 */
export class CreateItemUseCase {
  constructor(private readonly itemRepository: IItemRepository) {}

  /**
   * Execute the use case
   */
  async execute(data: CreateItemData): Promise<Item> {
    // Validate input
    const validatedData = ValidationService.validateCreateItem(data);

    // Apply defaults with proper enum casting
    const itemData = createItemWithDefaults({
      ...validatedData,
      condition: validatedData.condition as Condition,
      status: validatedData.status as Status
    });

    // Create item in repository
    const createdItem = await this.itemRepository.create(itemData);

    return createdItem;
  }
}
